src/TableFunctions/TableFunctionFile.cpp

```diff
@@ -58,6 +60,8 @@ String TableFunctionFile::getFormatFromFirstArgument()
 {
     if (fd >= 0)
         return FormatFactory::instance().getFormatFromFileDescriptor(fd);
+    else if (!filenames.empty())
+        return FormatFactory::instance().getFormatFromFileName(filenames.at(0), true);
```

Comment:
We should probably also check that the format is the same for all filenames, and throw an exception with `Only files with the same format are supported` if it is different. We should also check this case in tests.